### ISSUE TITLE
Optional EvalPort interop for LabelModel probabilistic labels

### ISSUE THREAD

**Issue by adhabnr-ux** [snorkel-team/snorkel#1737]
Hi Snorkel team — I maintain [EvalPort](https://github.com/adhabnr-ux/evalport), an open, framework-agnostic JSON spec for portable LLM eval datasets and results (a `TestCase`/`Suite`/`ResultSet` schema with a validator, so a dataset or a graded run can move between tools without hand-writing a converter each time). Filing this as an issue first per CONTRIBUTING.md before writing any code.

I installed `snorkel` (0.10.0) and ran the real API rather than guessing:

```python
applier = PandasLFApplier(lfs=[lf1, lf2, lf3])
L = applier.apply(df)
label_model = LabelModel(cardinality=2)
label_model.fit(L_train=L)
label_model.predict_proba(L)   # -> (n_examples, n_classes) float array
label_model.predict(L)         # -> per-example predicted label, -1 = abstain
LFAnalysis(L, lfs).lf_summary()  # -> DataFrame: j, Polarity, Coverage, Overlaps, Conflicts (+ Emp. Acc. if Y given)
```

`predict_proba`/`predict` give a per-example predicted label plus a real confidence score, and `lf_summary` gives per-labeling-function coverage/conflict/accuracy stats. That maps onto EvalPort's other half fairly directly: each row the LFs were applied to is a `TestCase`, and the `LabelModel`'s per-example (predicted label, probability) pair is exactly the shape of an EvalPort `Grader` result — a per-`TestCase` score/label — so a `ResultSet` could be built straight from `predict_proba` output without inventing any field names.

Two ways I could see this landing, and I don't have a strong preference:

1. **A standalone `snorkel-openeval-adapter` package** in the EvalPort repo, depending on `snorkel` as a normal dependency. Zero footprint on this repo.
2. **A small optional module inside this repo** if you'd rather it live here.

Either way, real tests would validate against EvalPort's actual JSON Schema, not a mock.

Let me know which direction you'd prefer, or if this isn't a fit for your roadmap right now — no worries either way.
